speech/src/main/java/com/example/speech/InfiniteStreamRecognize.java

```diff
@@ -285,18 +218,121 @@ public void onError(Throwable t) {}
 
             tempByteString = ByteString.copyFrom(sharedQueue.take());
 
+            checkStopRecognitionFlag(tempByteString.toByteArray());
+
             request =
                 StreamingRecognizeRequest.newBuilder().setAudioContent(tempByteString).build();
 
             audioInput.add(tempByteString);
           }
-
+          Thread.sleep(100);
```

Comment:
What is a reason for this delay? Why the delay is 100 ms? Code samples have to be self-explanatory. It is hard to understand from the code whether this delay is required and why 100 ms value is selected.